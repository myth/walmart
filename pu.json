{"questions": [{"description": "Hva er \u2018Software Engineering\u2019?", "answers": [{"description": "Er et sett av regler om hvordan man skal utvikle programvare.", "correct": false}, {"description": "Har v\u00e6rt en egen disiplin siden tidlig 50-tallet og er n\u00e5 en moden ingeni\u00f8rdisipli p\u00e5 linje med andre etablerte innenfor ingeni\u00f8rfeltet.", "correct": false}, {"description": "Startet som en respons p\u00e5 den s\u00e5kalte \u2018Software Crisis\u2019 sent p\u00e5 60-tallet.", "correct": false}, {"description": "Er en egen ingeni\u00f8rdisiplin som omfatter alle aspekter av programvareproduksjon", "correct": true}]}, {"description": "En programvareutvikler som tilslutter seg \u00abSoftware Engineering Code of Ethics and Professional Practices\u00bb m\u00e5:", "answers": [{"description": "Alltid \u00e5 praktisere i henhold til IEEE standarder", "correct": false}, {"description": "Hjelpe kollegaer i profesjonell utvikling", "correct": true}, {"description": "V\u00e6re medlem av IEEE Computer Society", "correct": false}, {"description": "Unng\u00e5 alle problemstillinger knyttet til finans og l\u00f8nnsomhet", "correct": false}]}, {"description": "En programvareingeni\u00f8r er ansatt i selskapet som designer og fremstiller avstemmingsmaskiner for bruk i kommune- og stortingsvalg. Hun er ansatt for \u00e5 utvikle arkivsystemet for \u00e5 sikre avstemmingsinformasjon og avstemningsresultater. Hvilken av de f\u00f8lgende handlinger vil mest sannsynlig v\u00e6re et brudd p\u00e5 \u00abSoftware Engineering Code of Ethics and Professional Practice\u00bb?", "answers": [{"description": "Hun unnlot \u00e5 levere korrekt og tilstrekkelig dokumentasjon av sitt arbeid i prosjektet.", "correct": true}, {"description": "Hun har ikke tidligere arbeidet i et prosjekt for utvikling av avstemmingsmaskiner.", "correct": false}, {"description": "Hun er registrert som stemmegiver i kommunen der maskinen vil bli benyttet.", "correct": false}, {"description": "Hun oppdaget et signifikant antall feil i eget design av arkivsystemet.", "correct": false}]}, {"description": "Hvilke av de f\u00f8lgende elementer skal IKKE v\u00e6re inkludert i en prosjektplan for utvikling av programvare?", "answers": [{"description": "Teknikkene og case-verkt\u00f8y som skal bli benyttet", "correct": false}, {"description": "Livssyklusmodellen som skal bli benyttet", "correct": false}, {"description": "Organisasjonsstruktur p\u00e5 utviklingsorganisasjonen, prosjektoppgaver, ledelsesm\u00e5l og prioriteter", "correct": false}, {"description": "Ingen av alternativene", "correct": true}]}, {"description": "En livssyklus for programvare kan bli forklart som \u00e5 best\u00e5 av en serie av faser. Den klassiske modellen blir kalt fossefallsmodellen. Hvilken fase kan defineres som 'hvor konsepter blir unders\u00f8kt og forfinet, og fremkaller derfor kundekrav?'", "answers": [{"description": "Krav", "correct": true}, {"description": "Spesifikasjon", "correct": false}, {"description": "Design", "correct": false}, {"description": "Implementering", "correct": false}]}, {"description": "Hva er det korrekte utsagnet i forhold til evolusjon\u00e6r utvikling/\u00abEvolutionary development\u00bb?", "answers": [{"description": "Evolusjon\u00e6r utvikling kommer normalt i to forskjellige former: Unders\u00f8kende/exploratory utvikling og bruk-og-kast prototyping", "correct": true}, {"description": "Veldig store prosjekter er naturlige kandidater til evolusjon\u00e6r utvikling", "correct": false}, {"description": "Unders\u00f8kende/exploratory utvikling blir benyttet i situasjoner der de fleste kravene p\u00e5 forh\u00e5nd.", "correct": false}, {"description": "Et av de sterkeste punktene for evolusjon\u00e6r utvikling er at det forenkler prosjektstyring gjennom den store mengden dokumentasjon som blir generert", "correct": false}]}, {"description": "I vedlikeholdsfasen m\u00e5 produktet testes mot tidligere test caser. Dette er ogs\u00e5 kjent som __________testing.", "answers": [{"description": "Enhets", "correct": false}, {"description": "Integrasjons", "correct": false}, {"description": "Regresjons", "correct": true}, {"description": "Akseptanse", "correct": false}]}, {"description": "Et eksempel p\u00e5 en risiko involvert i programvareutvikling:", "answers": [{"description": "Produsenten av kritiske komponenter (f.eks. maskinvare som er assosiert med et sanntidssystem) kan g\u00e5 konkurs.", "correct": false}, {"description": "Teknologiendringer kan medf\u00f8re at produktet blir foreldet", "correct": false}, {"description": "Konkurrenter kan markedsf\u00f8re et tilsvarende produkt til en lavere pris", "correct": false}, {"description": "Alle er riktig", "correct": true}]}, {"description": "Graden av interaksjon mellom to moduler er kjent som", "answers": [{"description": "Kohesjon/Cohesion", "correct": false}, {"description": "Arv/Inheritance", "correct": false}, {"description": "Kobling/Coupling", "correct": true}, {"description": "Instansiering/Instantiation.", "correct": false}]}, {"description": "Interne kostnader i et prosjekt inkluderer", "answers": [{"description": "L\u00f8nn til utviklere", "correct": true}, {"description": "L\u00f8nn til ledere og st\u00f8ttepersonell", "correct": false}, {"description": "Overheadkostnader f.eks. til hjelpesystemer, husleie og toppledelse", "correct": false}, {"description": "Materialer (som f.eks. manualer og b\u00f8ker) og tjenester som f.eks. reiser", "correct": false}]}, {"description": "Resultat fra algoritmisk kostnadsestimering i forskjellige organisasjoner kan v\u00e6re forskjellige for den samme type applikasjonsutvikling fordi", "answers": [{"description": "Forskjellige organisasjoner betrakter kompleksitetsfaktorer forskjellig og kan benytte forskjellige programmeringsspr\u00e5k", "correct": false}, {"description": "Dyktigheten til utviklere kan variere", "correct": false}, {"description": "Teknikker for m\u00e5ling av produktivitet kan variere", "correct": false}, {"description": "Alle er riktige.", "correct": true}]}, {"description": "Hvilke av elementene under fanger best naturen til et kvalitetsparadigme?", "answers": [{"description": "Kvalitetskarakteristikker, prosessperspektiv, eliminering av defekter", "correct": true}, {"description": "M\u00e5linger, kvalitetskontroll, validering", "correct": false}, {"description": "Gjennomf\u00f8rbarhet, krav, \u00f8konomi, brukerbehov", "correct": false}, {"description": "Analyse, test, design", "correct": false}]}, {"description": "N\u00e5r man velger en livssyklusmodell for programvareutvikling, s\u00e5 vil man ta i betraktning:", "answers": [{"description": "Ekspertisen til utviklingsgruppen, problemkarakteristikker, brukerforventninger", "correct": true}, {"description": "Programmeringsspr\u00e5k, tidsramme for utvikling, konkurransesituasjon", "correct": false}, {"description": "Systemkontekst, brukerpopulasjon, plattformer", "correct": false}, {"description": "Organisatorisk struktur, brukeroppgaver, ytelseskriteria", "correct": false}]}, {"description": "Hvilke av de f\u00f8lgende er IKKE betraktet som en prim\u00e6r drivkraft for \u00e5 forbedre programvareprosessen?", "answers": [{"description": "\u00d8kt effektivitet", "correct": false}, {"description": "Bedre produktkvalitet", "correct": false}, {"description": "Forbedring av personaltilfredshet", "correct": false}, {"description": "Tettere ledelseskontroll", "correct": true}]}, {"description": "N\u00e5r man skal planlegge et programvareprosjekt vil man:", "answers": [{"description": "Tilpasse tidsplanen for \u00e5 kunne h\u00e5ndtere eventuelle feil", "correct": false}, {"description": "Finne m\u00e5ter \u00e5 produsere resultater med tilstrekkelig kvalitet ved hjelp av begrensede ressurser.", "correct": true}, {"description": "Overestimere budsjettet (PI-faktor)", "correct": false}, {"description": "Alle svarene er riktig", "correct": false}]}, {"description": "Hvor mange tester m\u00e5 lages for \u00e5 teste fire betingelser i en IF-setning?", "answers": [{"description": "4", "correct": false}, {"description": "8", "correct": false}, {"description": "12", "correct": false}, {"description": "16", "correct": true}]}, {"description": "For \u00e5 v\u00e6re et effektivt verkt\u00f8y i prosessforbedring, s\u00e5 m\u00e5 underlagsmetrikkene v\u00e6re...", "answers": [{"description": "Basert p\u00e5 rimelige estimater fra mislykkede prosjekter", "correct": false}, {"description": "M\u00e5lt konsistent p\u00e5 tvers av prosjekter", "correct": true}, {"description": "Basert bare p\u00e5 vellykkede prosjekter", "correct": false}, {"description": "Basert p\u00e5 mislykkede prosjekter", "correct": false}]}, {"description": "En programvareutvikler m\u00e5ler kvaliteten av et programvaresystem. Hun er bekymret for p\u00e5liteligheten (\u00abreliability\u00bb) og validiteten (\u00abvalidity\u00bb) av m\u00e5lingene sine. Hvilken av f\u00f8lgende uttalelser er sann?", "answers": [{"description": "P\u00e5litelighet refererer til i hvilken grad m\u00e5lingene representerer den egentlige kvaliteten til systemet og validitet refererer til konsistensen av hennes kvalitetsm\u00e5linger.", "correct": false}, {"description": "P\u00e5litelighet refererer til konsistensen av hennes kvalitetsm\u00e5linger og validitet til i hvilken grad m\u00e5lingene representerer den egentlige kvaliteten til systemet.", "correct": true}, {"description": "P\u00e5litelighet refererer til n\u00f8yaktigheten p\u00e5 hennes kvalitetsm\u00e5linger og validitet referer til i hvilken grad m\u00e5lingene f\u00f8lger en kvalitetsstandard.", "correct": false}, {"description": "P\u00e5litelighet refererer til samtidigheten av hennes kvalitetsm\u00e5linger og validitet refererer til i hvilken grad m\u00e5lingene er konsistent med etablerte normer.", "correct": false}]}, {"description": "DIFI har definert sju arkitekturprinsipp. Seks av disse er tjenesteorientering, interoperabilitet, sikkerhet, \u00e5penhet, tilgjengelighet og skalerbarhet. Hva er det sjuende?", "answers": [{"description": "Fleksibilitet", "correct": true}, {"description": "Funksjonsstabilitet", "correct": false}, {"description": "Testbarhet", "correct": false}, {"description": "Ytelse", "correct": false}]}, {"description": "Interoperabilitet er et viktig arkitekturprinsipp. Hvilke typer interoperabilitet er definert av DIFI?", "answers": [{"description": "Obligatorisk bruk av web-protokollene SOAP, REST og JSON", "correct": false}, {"description": "Organisatorisk, semantisk og teknisk interoperabilitet", "correct": true}, {"description": "Etablering av felles IKT-l\u00f8sninger i offentlig sektor", "correct": false}, {"description": "Definerte standarder for kommunikasjon og integrasjon", "correct": false}]}], "subject": "PROGRAMVAREUTVIKLING"}