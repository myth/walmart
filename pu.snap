Programvareutvikling
Dependability is one of the essential attributes of good software.
..True
.False
Security is one of the essential attributes of good software.
..True
.False
The fundamental activities in software processes are: specification, development, and validation.
.True
..False
The software engineering code of ethics is a set of principles that prescribes, in a general way, standards of expected behaviour for professional software engineers.
..True
.False
In Incremental development, increments serve as examples to explore requirements.
..True
.False
In agile development, contract negotiation is of foremost importance.
.True
..False
In agile development, responding to change is more important than following a plan.
..True
.False
The principal requirements engineering activities are: Feasibility study; Requirements elicitation and analysis; Requirements validation.
.True
..False
Consider the following requirement specification fragment for a ticket-issuing system “Once a destination has been selected, users are requested to input their credit card. Its validity is checked and the user is then requested to input a personal identifier. When the credit transaction has been validated, the ticket is issued.” This is a functional requirement.
..True
.False
Consider the following requirement specification fragment for a ticket-issuing system “Between 0600 and 2300 in any one day, the recovery time after a system failure should not exceed 2 minutes”. This is an availability requirement.
..True
.False
Activity diagrams show object interactions arranged in time sequence.
.True
..False
A class diagram gives a view about the structural connections between classes in the system.
..True
.False
A textual use-case description describes: a) the actors involved; b) interactions; c) The data that is exchanged; d) The stimulus that triggers the use case; e) The response of the system; f) Comments.
..True
.False
According to Krutchen's 4+1 model, the logical view shows the key dynamic interactions of the system.
.True
..False
One of the aims of Configuration Management is to support system integration so that all developers can access the project code and documents in a controlled way.
..True
.False
Testdata is defined as input data used to execute the system during a test.
..True
.False
In open-source development, the source code of a software system is made publicly available.
..True
.False
In open-source development only volunteers and not professional developers can participate in the further development of the system.
.True
..False
Testing can detect the presence of errors as well as their absence.
.True
..False
During acceptance testing, customers test a system to check that it is ready for deployment.
..True
.False
New technologies which become available may impose that software systems must change to take advantage of them.
..True
.False
In general, effort of maintenance to adapt the software to a different environment can be estimated to be the 18% of all maintenance effort.
..True
.False
Program structure improvement is one possible re-engineering activity.
..True
.False
Communication ability is one of the important factors to consider when selecting people for a software project.
..True
.False
Project estimates must include either the effort required to complete each project activity or the time required to complete each activity.
.True
..False
